sbt-bridge/src/xsbt/CompilerClassLoader.scala

```diff
@@ -70,7 +80,10 @@ object CompilerClassLoader {
    *  @param bridgeLoader  The classloader that sbt uses to load the compiler bridge
    *  @return A fixed classloader that works with dotty
    */
-  def fixBridgeLoader(bridgeLoader: ClassLoader) = bridgeLoader match {
+  def fixBridgeLoader(bridgeLoader: ClassLoader): ClassLoader =
+    fixedLoaderCache.getOrElseUpdate(bridgeLoader, computeFixedLoader(bridgeLoader))
```

Comment:
You're right, this method should be synchronized. I don't think `loadClass` needs to be however because I'm only calling `super.loadClass` and `resolveClass` which should both be thread-safe. In fact, I think that I should be able to call https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#registerAsParallelCapable-- to make my ClassLoader parallel.